docs/guides/implementation-guides/ide-support.md

```diff
@@ -95,6 +95,19 @@ Alternatively you can use `Design.DataContext` attached property. As well as `De
 </Window>
 ```
 
+### Setting the Design-Time DataContext in Code
+
+Declaring the design-time `DataContext` in XAML will not work if the view model's only constructor has parameters. This happens with typical [dependency injection patterns](https://docs.avaloniaui.net/docs/guides/implementation-guides/how-to-implement-dependency-injection). One option is to use a special [design-time data model](https://docs.avaloniaui.net/docs/guides/implementation-guides/how-to-use-design-time-data) with a parameterless constructor.  However, a design-time `DataContext` that references a view model constructor with parameters may be set in code with the `Design.SetDataContext` method.
+
+```csharp
+public MainWindow()
+{
+    // This can be before or after InitializeComponent.
+    Design.SetDataContext(this, new MyViewModel(new DialogService()));
```

Comment:
Needs to be surrounded by an if statement - `if (Design.IsDesignMode)`. Otherwise, this is created while debugging the app and even in release mode. At least in my testing. It doesn't seem to happen if I set via XAML and I have no idea why.